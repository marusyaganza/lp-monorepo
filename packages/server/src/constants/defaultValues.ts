import { Language, Role, SortWordsBy, Tense } from '../generated/graphql';

export const WORDS_PER_PAGE = 5;
export const DEFAULT_LANGUAGE = Language.English;
export const DEFAULT_WORDS_SORT_BY = SortWordsBy.UpdatedAt;
export const DEFAULT_ROLE = Role.Member;
export const DEFAULT_TENSE = Tense.Pind;
export const DEFAULT_TOKEN_TTL = 7;
// to clean db every hour + 10 min just in case
export const DEMO_DB_TTL = 4200;
export const DEFAULT_MONGO_CONNECTION = 'mongodb://localhost:27017';
export const DEFAULT_OPTIONS__SAMPLE_SIZE = 50;
