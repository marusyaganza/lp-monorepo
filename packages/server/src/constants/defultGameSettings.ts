import { Game, Language, SettingsByGame } from '../generated/graphql';

export const DEFAULT_GAMES_SETTINGS: SettingsByGame = {
  [Game.Audio]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Image]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Speaking]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.TypeWord]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.SelectDef]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 6,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.SelectWord]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 6,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Conjugation]: {
    wordsPerGame: 2,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.Spanish]
  },
  [Game.Gender]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.Spanish]
  }
};
