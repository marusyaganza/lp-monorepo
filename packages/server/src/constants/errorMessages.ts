export const ERROR_MESSAGES = {
  NOT_AUTHENTICATED: 'Login to perform this operation',
  NOT_AUTHORIZED: 'You do not have permission to perform this operation',
  INCORRECT_CREDS: 'email or password is incorrect',
  SIGN_UP_FAILED: 'sign up operation failed',
  SING_UP_UNAVAILABLE:
    'Sign up operation is not available now. Sorry for inconvenience.',
  GAME_NOT_FOUND: 'Game not found',
  GAME_GENERATION_FAILED: 'Game data generation failed'
};
