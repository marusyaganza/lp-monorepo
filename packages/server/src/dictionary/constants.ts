import { Tense } from 'generated/graphql';

export type TagType = {
  tag: string;
  replacement: string;
};

export type ComplexTagType = {
  opening: string;
  closing: string;
};

export const TAGS: TagType[] = [
  { tag: '{b}', replacement: '<b>' },
  { tag: '{/b}', replacement: '</b>' },
  { tag: '{ldquo}', replacement: '\u201C' },
  { tag: '{rdquo}', replacement: '\u201D' },
  { tag: '{it}', replacement: '<i>' },
  { tag: '{/it}', replacement: '</i>' },
  { tag: '{wi}', replacement: '<i>' },
  { tag: '{/wi}', replacement: '</i>' },
  { tag: '{gloss}=', replacement: '<i>' },
  { tag: '{/gloss}', replacement: '</i>' },
  { tag: '{sc}', replacement: '<i>' },
  { tag: '{/sc}', replacement: '</i>' },
  { tag: '{parahw}', replacement: '</i>' },
  { tag: '{/parahw}', replacement: '</i>' },
  { tag: '{phrase}', replacement: '</i>' },
  { tag: '{/phrase}', replacement: '</i>' },
  { tag: '{qword}', replacement: '</i>' },
  { tag: '{/qword}', replacement: '</i>' },
  { tag: '{itsc}', replacement: '<i>' },
  { tag: '{/itsc}', replacement: '</i>' },
  { tag: '{inf}', replacement: '' },
  { tag: '{/inf}', replacement: '' },
  { tag: '{sup}', replacement: '' },
  { tag: '{/sup}', replacement: '' },
  { tag: '{bc}', replacement: '' },
  { tag: '{dx}', replacement: '' },
  { tag: '{/dx}', replacement: '' },
  { tag: '{dx_def}', replacement: '' },
  { tag: '{/dx_def}', replacement: '' },
  { tag: '{dx_ety}', replacement: '' },
  { tag: '{/dx_ety}', replacement: '' },
  { tag: '{ma}', replacement: '' },
  { tag: '{/ma}', replacement: '' },
  { tag: '{rom}', replacement: '' },
  { tag: '{/rom}', replacement: '' }
];

export const COMPLEX_TAGS: ComplexTagType[] = [
  { opening: '{sx', closing: '}' },
  { opening: '{d_link', closing: '}' },
  { opening: '{a_link', closing: '}' },
  { opening: '{i_link', closing: '}' },
  { opening: '{et_link', closing: '}' },
  { opening: '{mat', closing: '}' },
  { opening: '{dxt', closing: '}' }
];

export const COMPLEX_TAGS_TO_REMOVE: ComplexTagType[] = [
  { opening: '{ds|', closing: '}' }
];

export const HW_TAGS: TagType[] = [
  { tag: '{bit}', replacement: '' },
  { tag: '{/bit}', replacement: '' }
];

// Uncomment more tenses as needed
export const TENSES: Record<Tense, string> = {
  pind: 'Presente indicativo',
  pprf: 'Pretérito perfecto simple',
  impf: 'Imperativo',
  pret: 'Pretérito imperfecto'
  //   futr: 'Futuro indicativo',
  //   cond: 'Condicional',
  //   psub: 'Presente subjentivo',
  //   pisb1: 'Pretérito imperfecto',
  //   pisb2: 'Pretérito imperfecto 2',
  //   fsub: 'Futuro',
  //   ppci: 'Pretérito perfecto compuesto',
  //   ppsi: 'Pretérito pluscuamperfecto',
  //   pant: 'Pretérito anterior',
  //   fpin: 'Futuro perfecto',
  //   cpef: 'Condicional perfecto',
  //   ppfs: 'Pretérito perfecto',
  //   ppss1: 'Pretérito pluscuamperfecto',
  //   ppss2: 'Pretérito pluscuamperfecto 2',
  //   fpsb: 'Futuro perfecto',
};
