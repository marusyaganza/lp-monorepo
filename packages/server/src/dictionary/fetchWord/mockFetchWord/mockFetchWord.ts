import { Language } from '../../../generated/graphql';

export const mockSuggestions = {
  [Language.English]: [
    'Mocks are enabled. You can query words from the list: egalitarian, rubber, pussy, heart, wheel, voluminous, fowl, murther, or notFound to receive an empty array as a result, or suggestions for an array with words'
  ],
  [Language.Spanish]: [
    'Mocks are enabled. You can query words from the list: idioma, hola, caerse, asi, tener, ser, llover'
  ]
};

export async function mockFetchWord(
  query: string,
  language = Language.English
): Promise<unknown> {
  let result = [];
  const suggestions = mockSuggestions[language];
  if (!query || query === 'notFound') {
    return [];
  }

  try {
    const data = await import(`./mocks/${language}/${query}.js`);
    result = data?.default;
  } catch (err) {
    console.error('mockFetch err', err);
  }

  if (result?.length) {
    return result;
  }
  return suggestions;
}
