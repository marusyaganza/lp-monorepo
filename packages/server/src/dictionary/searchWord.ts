import { DictionaryWordType } from './types';
import { Language, SearchResult } from '../generated/graphql';
import { formatData } from './formatData';

const {
  DICTIONARY_API_URI,
  DICTIONARY_API_KEY,
  DICTIONARY_SPANISH_API_KEY,
  DICTIONARY_SPANISH_API_URI,
  DICTIONARY_THESAURUS_URI,
  DICTIONARY_THESAURUS_KEY
} = process.env;

const API = {
  spanish: {
    key: DICTIONARY_SPANISH_API_KEY,
    uri: DICTIONARY_SPANISH_API_URI
  },
  collegiate: {
    key: DICTIONARY_API_KEY,
    uri: DICTIONARY_API_URI
  },
  thesaurus: {
    key: DICTIONARY_THESAURUS_KEY,
    uri: DICTIONARY_THESAURUS_URI
  }
};

enum Dictionary {
  COLLEGIATE = 'collegiate',
  SPANISH = 'spanish'
}

const DICTIONARIES = {
  [Language.Spanish]: Dictionary.SPANISH,
  [Language.English]: Dictionary.COLLEGIATE
};

export async function fetchWord(
  query: string,
  dictionary: Dictionary
): Promise<DictionaryWordType[] | string[]> {
  let result: DictionaryWordType[] | string[] = [];
  if (!query) {
    return [];
  }
  const uri = `${API[dictionary].uri}/${query}?key=${API[dictionary].key}`;
  await fetch(uri)
    .then(res => res.json())
    .then(res => {
      result = res;
    })
    .catch(err => {
      console.error('dictionary fetch error', err);
    });
  return result;
}

export type SearchFuncType = (
  query: string,
  language: Language
) => Promise<SearchResult>;

export async function searchWord(
  query: string,
  language = Language.English
): Promise<SearchResult> {
  const words = await fetchWord(query, DICTIONARIES[language]);
  // @ts-ignore
  return formatData(words);
}
