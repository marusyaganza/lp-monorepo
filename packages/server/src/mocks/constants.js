export const mockEnv = {
  SALT_NUMBER: 3,
  JWT_SECTET: 'JWT_secret'
};
