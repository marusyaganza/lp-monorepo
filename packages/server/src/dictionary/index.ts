import { searchWord as searchWordFunc } from './searchWord';
import { mockSearchWord } from './mockSearchWord';

export const searchWord = process.env.USE_MOCKS
  ? mockSearchWord
  : searchWordFunc;
