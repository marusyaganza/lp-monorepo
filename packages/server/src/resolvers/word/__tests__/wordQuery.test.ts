import {
  createTestServer,
  getDataFromGQL,
  getErrorMessageFromGQL
} from '../../../tests/helpers';
import { words, user } from '../../../tests/mocks/data';
import { Language, SortWordsBy } from '../../../generated/graphql';
import { searchWord } from '../../../dictionary/searchWord';
import { models } from '../../../tests/mocks/mockModels';
import { wordQueries } from '../../../tests/mocks/gqlQueries';
import { ERROR_MESSAGES } from '../../../constants/errorMessages';
import { mockSuggestions } from '../../../dictionary/fetchWord/mockFetchWord/mockFetchWord';

describe('word queries', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.resetAllMocks();
  });
  test('wordsPerPage, English', async () => {
    const { query } = createTestServer({
      user,
      models
    });

    const input = {
      isReverseOrder: true,
      sortBy: SortWordsBy.UpdatedAt,
      language: Language.English
    };

    const res = await query({
      query: wordQueries.wordsQuery,
      variables: { input }
    });
    expect(getDataFromGQL(res)).toMatchSnapshot();
    expect(getErrorMessageFromGQL(res)).toEqual(undefined);
    expect(models.Word.findManyAndPaginate).toHaveBeenCalledTimes(1);
    expect(models.Word.findManyAndPaginate).toHaveBeenCalledWith(input, '1');
  });

  test('wordsPerPage, Spanish', async () => {
    const { query } = createTestServer({
      user,
      models
    });

    const input = { language: Language.Spanish };

    const res = await query({
      query: wordQueries.wordsQuery,
      variables: { input }
    });
    expect(getDataFromGQL(res)).toMatchSnapshot();
    expect(getErrorMessageFromGQL(res)).toEqual(undefined);
    expect(models.Word.findManyAndPaginate).toHaveBeenCalledWith(input, '1');
  });

  test('word by id', async () => {
    const { query } = createTestServer({
      user,
      models
    });

    const res = await query({
      query: wordQueries.wordByIdQuery,
      variables: { wordId: 'mockid' }
    });
    expect(getDataFromGQL(res)).toMatchSnapshot();
    expect(getErrorMessageFromGQL(res)).toEqual(undefined);
    expect(models.Word.findById).toHaveBeenCalledTimes(1);
  });

  test('word by id if word is not found', async () => {
    const { query } = createTestServer({
      user,
      models: {
        ...models,
        Word: {
          findById: jest.fn()
        }
      }
    });

    const res = await query({
      query: wordQueries.wordByIdQuery,
      variables: { wordId: 'mockid' }
    });
    expect(getErrorMessageFromGQL(res)).toEqual(
      'word with id mockid is not found'
    );
  });

  test('word by id if user is not found', async () => {
    const { query } = createTestServer({
      user,
      models: {
        ...models,
        User: {
          findOne: jest.fn()
        }
      }
    });

    const res = await query({
      query: wordQueries.wordByIdQuery,
      variables: { wordId: 'mockid' }
    });
    expect(getErrorMessageFromGQL(res)).toEqual(
      ERROR_MESSAGES.NOT_AUTHENTICATED
    );
  });

  test('words if user is not found', async () => {
    const { query } = createTestServer({
      user,
      models: {
        ...models,
        User: {
          findOne: jest.fn()
        }
      }
    });

    // @ts-expect-error: testing with invalid arguments
    const res = await query({
      query: wordQueries.wordsQuery
    });
    expect(getErrorMessageFromGQL(res)).toEqual(
      ERROR_MESSAGES.NOT_AUTHENTICATED
    );
  });

  test('searchWord if user is not found', async () => {
    const { query } = createTestServer({
      user,
      models: {
        ...models,
        User: {
          findOne: jest.fn()
        }
      }
    });

    const res = await query({
      query: wordQueries.searchQuery,
      variables: {
        input: {
          search: 'word',
          language: Language.English
        }
      }
    });
    expect(getErrorMessageFromGQL(res)).toEqual(
      ERROR_MESSAGES.NOT_AUTHENTICATED
    );
  });

  test('searchWord', async () => {
    const searchWord = jest.fn(() => words);
    const { query } = createTestServer({
      user,
      searchWord,
      models
    });
    await query({
      query: wordQueries.searchQuery,
      variables: {
        input: {
          search: 'word',
          language: Language.English
        }
      }
    });
    expect(searchWord).toHaveBeenCalledWith('word', Language.English);
  });

  test('searchWord spanish lang, mocks enabled', async () => {
    const { query } = createTestServer({
      user,
      searchWord,
      models
    });
    const res = await query({
      query: wordQueries.searchQuery,
      variables: {
        input: {
          search: 'idioma',
          language: Language.Spanish
        }
      }
    });
    expect(getDataFromGQL(res)).toMatchSnapshot();
    expect(getErrorMessageFromGQL(res)).toEqual(undefined);
  });

  test('searchWord english lang, mocks enabled', async () => {
    const { query } = createTestServer({
      user,
      searchWord,
      models
    });
    const res = await query({
      query: wordQueries.searchQuery,
      variables: {
        input: {
          search: 'rubber',
          language: Language.English
        }
      }
    });
    expect(getDataFromGQL(res)).toMatchSnapshot();
    expect(getErrorMessageFromGQL(res)).toEqual(undefined);
  });

  test('searchWord spanish lang, mocks enabled returns suggestion', async () => {
    const { query } = createTestServer({
      user,
      searchWord,
      models
    });
    const res = await query({
      query: wordQueries.searchQuery,
      variables: {
        input: {
          search: 'unbelievableWordThatDoNotExist',
          language: Language.Spanish
        }
      }
    });
    expect(getDataFromGQL(res).searchWord[0].suggestions).toEqual(
      mockSuggestions[Language.Spanish]
    );
  });

  test('searchWord english lang, mocks enabled returns suggestion', async () => {
    const { query } = createTestServer({
      user,
      searchWord,
      models
    });
    const res = await query({
      query: wordQueries.searchQuery,
      variables: {
        input: {
          search: 'unbelievableWordThatDoNotExist',
          language: Language.English
        }
      }
    });
    expect(getDataFromGQL(res).searchWord[0].suggestions).toEqual(
      mockSuggestions[Language.English]
    );
  });
});
