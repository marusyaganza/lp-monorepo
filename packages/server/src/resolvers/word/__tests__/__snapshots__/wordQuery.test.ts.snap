// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`word queries searchWord english lang, mocks enabled 1`] = `
{
  "searchWord": [
    {
      "additionalInfo": null,
      "audioUrl": "mock_audio_endpoint/en/us/mp3/r/rubber01.mp3",
      "defs": [
        {
          "def": "a strong substance that stretches and that is made out of chemicals or from the juice of a tropical tree, see also foam rubber",
          "examples": [
            {
              "text": "tires made of <i>rubber</i>",
              "translation": null,
            },
          ],
        },
        {
          "def": "eraser",
          "examples": null,
        },
        {
          "def": "shoes or boots that are made of rubber and that fit over your regular shoes to keep them dry compare galoshes",
          "examples": null,
        },
        {
          "def": "condom",
          "examples": null,
        },
        {
          "def": "a flat piece of hard, white rubber on which the pitcher stands while pitching",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "rubber",
      "particle": "noun",
      "shortDef": [
        "a strong substance that stretches and that is made out of chemicals or from the juice of a tropical tree —often used before another noun",
        "eraser",
        "shoes or boots that are made of rubber and that fit over your regular shoes to keep them dry",
      ],
      "stems": [
        "rubber",
        "rubbers",
        "burn rubber",
      ],
      "transcription": "rub*ber",
      "uuid": "59e03286-73c9-43e6-b1eb-4407a6c06b1e",
    },
    {
      "additionalInfo": null,
      "audioUrl": null,
      "defs": [
        {
          "def": "a contest that consists of a series of games and that is won by the player or team that wins the most games (such as two games out of three)",
          "examples": [
            {
              "text": "Let's play a <i>rubber</i> of bridge.",
              "translation": null,
            },
          ],
        },
        {
          "def": "a game that decides who will be the winner of a series of games",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "rubber",
      "particle": "noun",
      "shortDef": [
        "a contest that consists of a series of games and that is won by the player or team that wins the most games (such as two games out of three)",
        "a game that decides who will be the winner of a series of games—usually used before another noun",
      ],
      "stems": [
        "rubber",
        "caoutchouc",
        "india rubber",
        "rubbers",
        "rubber game",
      ],
      "transcription": "rubber",
      "uuid": "31c81e56-7a47-40ee-b6b3-63407105feb4",
    },
    {
      "additionalInfo": null,
      "audioUrl": "mock_audio_endpoint/en/us/mp3/c/condom01.mp3",
      "defs": [
        {
          "def": "a thin rubber covering that a man wears on his penis during sex in order to prevent a woman from becoming pregnant or to prevent the spread of diseases",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "condom",
      "particle": "noun",
      "shortDef": [
        "a thin rubber covering that a man wears on his penis during sex in order to prevent a woman from becoming pregnant or to prevent the spread of diseases —called also rubber, sheath",
      ],
      "stems": [
        "condom",
        "condoms",
        "rubber",
        "sheath",
      ],
      "transcription": "con*dom",
      "uuid": "79b4d0fe-8d9d-4bd0-879c-366078d586d6",
    },
    {
      "additionalInfo": null,
      "audioUrl": "mock_audio_endpoint/en/us/mp3/e/eraser01.mp3",
      "defs": [
        {
          "def": "a small piece of rubber or other material that is used to erase something you have written or drawn, see picture at office",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "eraser",
      "particle": "noun",
      "shortDef": [
        "a small piece of rubber or other material that is used to erase something you have written or drawn —called also (British) rubber",
      ],
      "stems": [
        "eraser",
        "erasers",
        "rubber",
      ],
      "transcription": "eras*er",
      "uuid": "eb991c0d-a218-4812-930d-9ea376c44445",
    },
    {
      "additionalInfo": null,
      "audioUrl": null,
      "defs": [
        {
          "def": "a thin, flexible loop that is made of rubber and used to hold things together, see picture at office",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "rubber band",
      "particle": "noun",
      "shortDef": [
        "a thin, flexible loop that is made of rubber and used to hold things together —called also (US) elastic, elastic band",
      ],
      "stems": [
        "rubber band",
        "rubber bands",
        "elastic",
        "elastic band",
      ],
      "transcription": "rubber band",
      "uuid": "0634ec1c-dc66-465f-b390-fb945313d066",
    },
    {
      "additionalInfo": null,
      "audioUrl": null,
      "defs": [
        {
          "def": "a tall boot that is made out of rubber and that keeps your feet and lower legs dry",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "rubber boot",
      "particle": "noun",
      "shortDef": [
        "a tall boot that is made out of rubber and that keeps your feet and lower legs dry —called also (British) Wellington",
      ],
      "stems": [
        "rubber boot",
        "rubber boots",
        "wellington",
      ],
      "transcription": "rubber boot",
      "uuid": "48d907a0-a8fb-4ca3-bc75-57a70d908166",
    },
    {
      "additionalInfo": null,
      "audioUrl": null,
      "defs": [
        {
          "def": "a bullet that is made out of rubber so that it hurts but does not kill people and that is used by the police and military to control a crowd",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "rubber bullet",
      "particle": "noun",
      "shortDef": [
        "a bullet that is made out of rubber so that it hurts but does not kill people and that is used by the police and military to control a crowd",
      ],
      "stems": [
        "rubber bullet",
        "rubber bullets",
      ],
      "transcription": "rubber bullet",
      "uuid": "dbdc0d78-3e16-4dc3-afd5-2e72bf8b6a55",
    },
    {
      "additionalInfo": null,
      "audioUrl": null,
      "defs": [
        {
          "def": "a thick, stretchy glue that contains rubber",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "rubber cement",
      "particle": "noun",
      "shortDef": [
        "a thick, stretchy glue that contains rubber",
      ],
      "stems": [
        "rubber cement",
        "rubber cements",
      ],
      "transcription": "rubber cement",
      "uuid": "eb1c2aba-f3df-46d5-936b-626e499998b1",
    },
    {
      "additionalInfo": null,
      "audioUrl": null,
      "defs": [
        {
          "def": "a plant with large, shiny, dark green leaves that is often grown indoors",
          "examples": null,
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "rubber plant",
      "particle": "noun",
      "shortDef": [
        "a plant with large, shiny, dark green leaves that is often grown indoors",
      ],
      "stems": [
        "rubber plant",
        "rubber plants",
      ],
      "transcription": "rubber plant",
      "uuid": "39f6a4ec-4959-4d6f-876a-39b8f780c8cd",
    },
    {
      "additionalInfo": null,
      "audioUrl": null,
      "defs": [
        {
          "def": "a small tool that consists of a handle attached to a piece of carved rubber and that is dipped in ink and then pressed on paper to print a date, name, symbol, etc.",
          "examples": null,
        },
        {
          "def": "a person or organization that automatically approves everything that someone does or decides",
          "examples": [
            {
              "text": "The legislature has been nothing more than a <i>rubber stamp</i> for the President.",
              "translation": null,
            },
          ],
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "ENGLISH",
      "name": "rubber stamp",
      "particle": "noun",
      "shortDef": [
        "a small tool that consists of a handle attached to a piece of carved rubber and that is dipped in ink and then pressed on paper to print a date, name, symbol, etc.",
        "a person or organization that automatically approves everything that someone does or decides",
      ],
      "stems": [
        "rubber stamp",
        "rubber stamps",
        "rubber-stamp",
      ],
      "transcription": "rubber stamp",
      "uuid": "3d83c22d-0913-4b18-8462-a4064a7d8fd2",
    },
  ],
}
`;

exports[`word queries searchWord spanish lang, mocks enabled 1`] = `
{
  "searchWord": [
    {
      "additionalInfo": null,
      "audioUrl": "mock_audio_endpoint/es/me/mp3/i/idiom01sp.mp3",
      "defs": [
        {
          "def": "language",
          "examples": [
            {
              "text": "el idioma inglés",
              "translation": "the English language",
            },
          ],
        },
      ],
      "imgDesc": null,
      "imgUrl": null,
      "isOffensive": false,
      "language": "SPANISH",
      "name": "idioma",
      "particle": "masculine noun",
      "shortDef": [
        "language",
      ],
      "stems": [
        "idioma",
      ],
      "transcription": "idioma",
      "uuid": "63882a81-032e-4dc1-9c22-bb683cb1bad9",
    },
  ],
}
`;

exports[`word queries word by id 1`] = `
{
  "word": {
    "additionalInfo": null,
    "audioUrl": "https://www.oxfordlearnersdictionaries.com/media/english/uk_pron/e/ega/egali/egalitarian__gb_1.mp3",
    "defs": [
      {
        "def": "a person who believes that everyone is equal and should have the same rights and opportunities",
        "examples": [
          {
            "text": "He described himself as ‘an {it}egalitarian{/it}’.",
            "translation": null,
          },
        ],
      },
      {
        "def": "aiming for equal wealth, status, etc., for all people",
        "examples": [
          {
            "text": "He is a committed {it}egalitarian{/it}",
            "translation": null,
          },
          {
            "text": "Even in the {it}egalitarian{/it} Nordic country, Marin felt hergender and age sometimes received too much emphasis. Kostya Manenkov And Karl Ritter, USA TODAY, 21 Aug. 2022",
            "translation": null,
          },
        ],
      },
    ],
    "id": "mockid",
    "imgDesc": null,
    "imgUrl": null,
    "isLearned": null,
    "isOffensive": false,
    "level": "C1",
    "name": "egalitarian",
    "particle": "noun",
    "shortDef": [
      "asserting, promoting, or marked by egalitarianism",
    ],
    "stems": [
      "egalitarian",
      "egalitarianism",
      "egalitarians",
    ],
    "tags": null,
    "transcription": "iˌɡælɪˈteəriən",
  },
}
`;

exports[`word queries wordsPerPage, English 1`] = `
{
  "wordsPerPage": {
    "hasNext": false,
    "words": [
      {
        "audioUrl": "",
        "defs": [
          {
            "def": "an African bird (<i>Numida meleagris</i>) related to the pheasants, raised for food in many parts of the world, and marked by a bare neck and head and slaty plumage speckled with white",
          },
        ],
        "id": "66f7ef03eae42983ce978474",
        "isLearned": false,
        "isOffensive": false,
        "level": "B1",
        "name": "guinea fowl",
        "particle": "noun",
        "shortDef": [
          "an African bird (Numida meleagris) related to the pheasants, raised for food in many parts of the world, and marked by a bare neck and head and slaty plumage speckled with white; broadly : any of several related birds",
        ],
        "tags": [
          {
            "color": "#56c8a2",
            "id": "66f7ef1ceae42983ce978482",
            "text": "animals",
          },
        ],
        "transcription": "guinea fowl",
      },
      {
        "audioUrl": "https://media.merriam-webster.com/audio/prons/en/us/mp3/f/fish0001.mp3",
        "defs": [
          {
            "def": "any of numerous cold-blooded strictly aquatic craniate vertebrates that include the bony fishes and usually the cartilaginous, cartilaginous fish and jawless fishes and that have typically an elongated somewhat spindle-shaped body terminating in a broad caudal see caudal fin, limbs in the form of fins when present at all, and a 2-chambered heart by which blood is sent through thoracic gills to be oxygenated",
          },
          {
            "def": "the flesh of fish used as food",
          },
          {
            "def": "sucker",
          },
          {
            "def": "torpedo",
          },
        ],
        "id": "66f7ef05eae42983ce978478",
        "isLearned": null,
        "isOffensive": false,
        "level": null,
        "name": "fish",
        "particle": "noun",
        "shortDef": [
          "an aquatic animal —usually used in combination",
          "any of numerous cold-blooded strictly aquatic craniate vertebrates that include the bony fishes and usually the cartilaginous and jawless fishes and that have typically an elongated somewhat spindle-shaped body terminating in a broad caudal fin, limbs in the form of fins when present at all, and a 2-chambered heart by which blood is sent through thoracic gills to be oxygenated",
          "the flesh of fish used as food",
        ],
        "tags": [],
        "transcription": "ˈfish",
      },
      {
        "audioUrl": "https://media.merriam-webster.com/audio/prons/en/us/mp3/m/murthe01.mp3",
        "defs": [
          {
            "def": "chiefly dialectal variant of <i>murder</i>",
          },
        ],
        "id": "66f7daa8ac5cc9e644a97773",
        "isLearned": null,
        "isOffensive": false,
        "level": null,
        "name": "murther",
        "particle": "noun",
        "shortDef": [
          "chiefly dialectal variant of <i>murder</i>",
        ],
        "tags": [],
        "transcription": "ˈmər-t͟hər",
      },
    ],
    "wordsCount": 3,
  },
}
`;

exports[`word queries wordsPerPage, Spanish 1`] = `
{
  "wordsPerPage": {
    "hasNext": false,
    "words": [
      {
        "audioUrl": null,
        "defs": [
          {
            "def": "to fall",
          },
        ],
        "id": "0",
        "isLearned": null,
        "isOffensive": null,
        "level": null,
        "name": "caer",
        "particle": "intransitive verb",
        "shortDef": [
          "to fall",
          "to drop away, to slope",
          "to fall (of night)",
        ],
        "tags": null,
        "transcription": null,
      },
    ],
    "wordsCount": 1,
  },
}
`;
